src/server/services/wms/qgswmsgetcapabilities.cpp

```diff
@@ -624,33 +624,34 @@ namespace QgsWms
 
   QDomElement getComposerTemplatesElement( QDomDocument &doc, const QgsProject *project )
   {
-    QList<QgsComposition *> projectComposers = project->layoutManager()->compositions();
+    QList< QgsPrintLayout * > projectComposers = project->layoutManager()->printLayouts();
     if ( projectComposers.size() == 0 )
       return QDomElement();
 
     QStringList restrictedComposers = QgsServerProjectUtils::wmsRestrictedComposers( *project );
 
     QDomElement composerTemplatesElem = doc.createElement( QStringLiteral( "ComposerTemplates" ) );
-    QList<QgsComposition *>::const_iterator cIt = projectComposers.constBegin();
+    QList<QgsPrintLayout *>::const_iterator cIt = projectComposers.constBegin();
     for ( ; cIt != projectComposers.constEnd(); ++cIt )
     {
-      QgsComposition *composer = *cIt;
-      if ( restrictedComposers.contains( composer->name() ) )
+      QgsPrintLayout *layout = *cIt;
+      if ( restrictedComposers.contains( layout->name() ) )
         continue;
 
       QDomElement composerTemplateElem = doc.createElement( QStringLiteral( "ComposerTemplate" ) );
-      composerTemplateElem.setAttribute( QStringLiteral( "name" ), composer->name() );
+      composerTemplateElem.setAttribute( QStringLiteral( "name" ), layout->name() );
 
       //get paper width and hight in mm from composition
-      composerTemplateElem.setAttribute( QStringLiteral( "width" ), composer->paperWidth() );
-      composerTemplateElem.setAttribute( QStringLiteral( "height" ), composer->paperHeight() );
+      composerTemplateElem.setAttribute( QStringLiteral( "width" ), layout->width() );
```

Comment:
You'll need to take the first page size here - the width and height functions are unfortunately inherited from QGraphicsScene and give the whole scene bounds dimensions. This includes all items, including cosmetic things like guides, page shadow, etc.